src/entries/popup/components/TestnetMode/TestnetModeWatcher/TestnetModeWatcher.tsx

```diff
@@ -0,0 +1,178 @@
+import { TextLink } from '~/design-system/components/TextLink/TextLink';
+import { useAppSession } from '~/entries/popup/hooks/useAppSession';
+import { zIndexes } from '~/entries/popup/utils/zIndexes';
+
+import { ChainBadge } from '../../ChainBadge/ChainBadge';
+import { Navbar } from '../../Navbar/Navbar';
+
+type Hint = {
+  show: boolean;
+  type: 'tesnetModeInMainnet' | 'notTestnetModeInTestnet';
+  chainId: ChainId;
+};
+
+const INITIAL_HINT: Hint = {
+  show: false,
+  type: 'tesnetModeInMainnet',
+  chainId: ChainId.mainnet,
+};
+
+export const TestnetModeWatcher = ({
+  pendingRequest,
+  rejectRequest,
+}: {
+  pendingRequest?: ProviderRequestPayload;
+  rejectRequest?: () => void;
+}) => {
+  const { data: dappMetadata } = useDappMetadata({
+    url: pendingRequest?.meta?.sender.url,
+  });
+  const dappHost = dappMetadata?.appHost || '';
+
+  const { testnetMode, setTestnetMode } = useTestnetModeStore();
+  const { activeSession } = useAppSession({
+    host: dappHost,
+  });
+
+  const [hint, setHint] = useState<Hint>(INITIAL_HINT);
+
+  const closeSheet = () => {
+    setHint(INITIAL_HINT);
+    rejectRequest?.();
+  };
+
+  const action = () => {
+    setHint(INITIAL_HINT);
+    setTestnetMode(!testnetMode);
+  };
+
+  useEffect(() => {
+    if (activeSession && !hint.show) {
+      const activeSessionChainId = activeSession?.chainId;
+      const activeChainIsTestnet = isTestnetChainId({
+        chainId: activeSessionChainId,
+      });
+      if (testnetMode && !activeChainIsTestnet) {
+        setHint({
+          show: true,
+          type: 'tesnetModeInMainnet',
+          chainId: activeSessionChainId,
+        });
+      } else if (!testnetMode && activeChainIsTestnet) {
+        setHint({
+          show: true,
+          type: 'notTestnetModeInTestnet',
+          chainId: activeSessionChainId,
+        });
+      }
+    }
+  }, [activeSession, hint.show, testnetMode]);
+
+  return (
+    <BottomSheet show={hint.show} zIndex={zIndexes.BOTTOM_SHEET}>
+      <Box>
+        <Navbar
+          leftComponent={
+            <Navbar.CloseButton variant="transparent" onClick={closeSheet} />
+          }
+        />
+        <Box marginTop="-16px" paddingHorizontal="20px">
+          <Stack space="24px" alignHorizontal="center">
+            <ChainBadge size="45" chainId={hint.chainId} />
+            <Stack space="12px" alignHorizontal="center">
+              <Text color="label" size="16pt" weight="bold">
+                {i18n.t('testnet_mode_watcher.connect_to', {
+                  chainName: ChainNameDisplay[hint.chainId],
+                })}
+              </Text>
+              <Text
+                align="center"
+                color="labelTertiary"
+                size="10pt"
+                weight="bold"
+              >
+                🕹️{' '}
+                <TextLink scale={false} color="green">
+                  {i18n.t('testnet_mode_watcher.testnet_mode')}
+                </TextLink>{' '}
```

Comment:
same here

Reply:
but not this, will fix thanks

Reply:
actually, i do want it, is a link inside a text